### DISCUSSION TITLE
Add free IP addresses

### DISCUSSION THREAD

**Discussion by qsfpdd** [netbox-community/netbox#12944] [Help Wanted!] [answered]
Hello, is there any way to add to IPAM all available IP addresses but with another status (not "Active", maybe "Available") which will not be counted to range's utilization?
Thank you

**Comment by pobradovic08** [comment]
I'm not aware of a way to have an IP created but not be counted as a used IP. Can you explain your use case, why would you create IP address objects for free IPs?

The idea is not to have IP address objects created for available IPs, so that NetBox will assign you the next available IP when you need it. If the IP is available, then you don't have any knowledge about its future use that you can put into IPAM. If you do have some information about it's future use, then that information should probably be in form of:
- A reservation (and count toward used space as the IP is reserved and can't be used for anything else) if we're talking about just one IP
- An IP range for stuff like DHCP or customer allocated pool
- The whole prefix

**Comment by qsfpdd** [comment]
I wanted to also add available IPs in order to be able to find quickly an available address when needed too.  It would be nice if there was a list with all free IPs of a range.
Regarding NetBox assigning itself the IPs, are you referring to API calls?

**Reply by stavr666** [reply, marked as answer]
Over UI also works.
Go to "Prefixes" module, find prefix, where you gonna assign new address, select "IP Addresses" tab (path string looks like /ipam/prefixes/<prefix_id>/ip-addresses/), press "+ Add IP Address" in upper right corner:
![image](https://github.com/netbox-community/netbox/assets/84839985/0d68541d-c176-4845-8f0b-17e7b2faa708)

Your new IP will be outside of existing IPs and defined ranges inside Prefix.

**Reply by qsfpdd** [reply]
You are right, I found it in the path you said. To be honest I was working with Ranges; Prefixes seem much better. Thank you very much!
